File: DictationAssistant/WaitingTimeBasedOnWordCount.vb

''' <summary>
''' 播报间隔接口
''' </summary>
''' <remarks></remarks>
''' <summary>
''' Seconds = SecondsPerWord * WordCount + BasedSecond
''' </summary>
Public Class WaitingTimeBasedOnWordCount
    Implements IWaitingTimeCalculator
    Private SecondsPerWord As Int32
    Private BasedSecond As Int32
    Public Sub New(SecondsPerWord As Integer, Optional BasedSecond As Integer = 0)
        Me.SecondsPerWord = SecondsPerWord
        Me.BasedSecond = BasedSecond
    End Sub
    Public Function CalculateWaitingTime(Word As String) As Int32 Implements IWaitingTimeCalculator.CalculateWaitingTime
        Return If(SecondsPerWord <> 0, WaitingTimeHelper.GetWordCount(Word) * SecondsPerWord, 0) + BasedSecond
    End Function
End Class

File: DictationAssistant/WaitingTimeBasedOnWordCount.cs
using System;

namespace DictationAssistant
{
    public class WaitingTimeBasedOnWordCount : IWaitingTimeCalculator
    {
        private Int32 SecondsPerWord;
        private Int32 BasedSecond;
        public WaitingTimeBasedOnWordCount(int secondsPerWord, int basedSecond = 0)
        {
            this.SecondsPerWord = secondsPerWord;
            this.BasedSecond = basedSecond;
        }
        public Int32 CalculateWaitingTime(string word)
        {
            return SecondsPerWord != 0 ? WaitingTimeHelper.GetWordCount(word) * SecondsPerWord : 0 + BasedSecond;
        }
    }
}
